### PR TITLE
TINKERPOP-3081 Fix traversal argument propagation under authentication

### PR DESCRIPTION
**Pull request by lyndonbauto**
Issue is that when using authentication traversal arguments  (for example evaluationTimeout do not propagate, leading to failures in the traversal and unexpected behavior. This fixes that problem

### CODE REVIEW

**Comment by vkagamlyk** [comment]
Good to have better PR description.
Wrong target brunch.

**Comment by vkagamlyk** [comment]
VOTE+1

File: gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/WebSocketAuthorizationHandler.java [hunk 1]
```diff
@@ -68,11 +68,19 @@ public void channelRead(final ChannelHandlerContext ctx, final Object msg) {
                         final Bytecode bytecode = (Bytecode) requestMessage.getArgs().get(Tokens.ARGS_GREMLIN);
                         final Map<String, String> aliases = (Map<String, String>) requestMessage.getArgs().get(Tokens.ARGS_ALIASES);
                         final Bytecode restrictedBytecode = authorizer.authorize(user, bytecode, aliases);
-                        final RequestMessage restrictedMsg = RequestMessage.build(Tokens.OPS_BYTECODE).
+                        final RequestMessage.Builder restrictedMsgBuilder = RequestMessage.build(Tokens.OPS_BYTECODE).
```
Comment (kenhuuu) [inline gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/WebSocketAuthorizationHandler.java:71]: I think you could clean this up a little by using the from() (https://github.com/apache/tinkerpop/blob/3.6-dev/gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/message/RequestMessage.java#L107) to make a copy of the old RequestMessage and then just override the ARGS_GREMLIN with restrictedBytecode.

Comment (lyndonbauto) [reply gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/WebSocketAuthorizationHandler.java]: Updated

**Comment by kenhuuu** [comment]
Might be good to also add an integration test for this. Probably here https://github.com/apache/tinkerpop/blob/3.6-dev/gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinServerAuthzIntegrateTest.java

**Comment by lyndonbauto** [comment]
@kenhuuu okay will do, might be a bit im just wrappedu p in some other stuff with work but this is in the back of my mind.

**Comment by kenhuuu** [comment]
VOTE +1

**Comment by xiazcy** [comment]
Could you check your test set up? It seems like there is a serialization error with it. 
`org.apache.tinkerpop.gremlin.driver.ser.SerializationException: java.io.IOException: Serializer for type org.apache.tinkerpop.gremlin.server.GremlinServerAuthzIntegrateTest$$Lambda$597/0x000000080074d840 not found`

**Comment by lyndonbauto** [comment]
@xiazcy Yeah its wrong right now, I need to do it a different way, I wasn't able to test locally due to jvm installation mismatches so I tried that blindly. I'll try to see if I can do it a different way.

**Comment by xiazcy** [comment]
VOTE +1

**Comment by kenhuuu** [comment]
Hi Lyndon, were you planning on merging this? If not, I can help you merge it this week. I'd prefer not to leave this PR hanging around since it might cause conflicts with other PRs (the CHANGELOG, in particular, usually causes conflicts).

**Comment by lyndonbauto** [comment]
@kenhuuu go ahead and merge it, sorry I am caught up in some stuff right now

**Pull request merged**